File: SimpleDownloadManager/DownloadFile.m
//
//  DownloadFIle.m
//  SimpleDownloadManager
//
//  Created by Sam on 11/11/16.
//  Copyright © 2016 Roma. All rights reserved.
//

#import "DownloadFile.h"

@implementation DownloadFile


-(instancetype) initWithTitleName:(NSString*)titleName
                       urlString:(NSString*)urlString;
{
    
        self = [super init];
        if (self) {
            self.titleName  = titleName;
            self.url       = [NSURL URLWithString:urlString];
        }
    return self;
}







@end

File: SimpleDM/DownloadFile.swift
//
//  DownloadFile.swift
//  SimpleDM
//
//  Created by Sam on 11/21/16.
//  Copyright © 2016 Roma. All rights reserved.
//

import UIKit

typealias downloadProgressBlock = (Float,String)->()
typealias isCompletedDownloadingBlock = (Bool)->();
typealias downloadTimeBlock = (String)->();


class DownloadFile: NSObject {
    
    internal var downloadTask:URLSessionDownloadTask?;
    
    let titleName:String!;
    let url:URL!;
    
     var dateStartDownload:NSDate?;
     var dateCompletDownload:NSDate?;
     var statusCompleted:String?;
    
     var isInit = false
    
    var progressBlock:downloadProgressBlock?;
    var isCompletedDownloadingBlock:isCompletedDownloadingBlock?;
    var downloadTimeBlock:downloadTimeBlock?;

    
    init (titleName:String, urlString:String)
    {
        self.titleName = titleName;
        self.url = URL(string: urlString)!;
    
    }
    
   
}
